README.md

```diff
@@ -1971,4 +1971,244 @@ You can create a `./gradlew checkstyle` command using the following:
 task checkstyle {
     dependsOn(checkstyleMain, checkstyleTest)
 }
-```
\ No newline at end of file
+```
+
+## TDD with Fibonacci
+
+As with all Test Driven Development, the key things to remember are:
+
+1. Red: Write as little code as possible to make your test fail.
+1. Green: Write as little code as possible to make your test pass.
+1. Refactor: Optimise your code for readability without changing its behaviour.
+
+First, create a test class that calls the intended target class:
+
+``` java
+public class FibonacciTest {
+    
+    Fibonacci target = new Fibonacci();
+
+}
+```
+
+Then create that target class.
+
+``` java
+public class Fibonacci {
+}
+```
+
+Second, add a simple method to the test class.
+
+``` java
+public class FibonacciTest {
+
+    public void shouldBe() {
+        Fibonacci target = new Fibonacci();
+        target.get(0);
+    }
+
+}
+```
+
+Then create the method `get` in the target class.
+
+``` java
+public class Fibonacci {
+
+    public int get() {
+        return 0;
+    }
+    
+}
+```
+
+Third, create a simple test in the test class.
+
+``` java
+import static org.junit.Assert.assertEquals;
+
+public class FibonacciTest {
+
+    @Test
+    public void shouldBe() {
+        Fibonacci target = new Fibonacci();
+        assertEquals(1,target.get());
+    }
+
+}
+```
+
+Then modify the method in the target class.
+
+``` java
+public class Fibonacci {
+
+    public int get() {
+        return 1;
+    }
+    
+}
+```
+
+Fourth, add more tests/assertions to the test class.
+
+``` java
+import static org.junit.Assert.assertEquals;
+
+public class FibonacciTest {
+
+    @Test
+    public void shouldBe() {
+        Fibonacci target = new Fibonacci();
+        assertEquals(1,target.get());
+        assertEquals(3,target.get());
+    }
+
+}
+```
+
+Then update the target class.
+
+``` java
+public class Fibonacci {
+
+    public int get(int n) {
+        if (n < 3) {
+            return 1;
+        }
+        return 2;
+    }
+}
+```
+
+There is now a lot of refactoring that can be done to the test class.
+
+A target field can be created, as can a private method called `shouldBe()`.
+
+Then further tests/assertions can be added.
+
+``` java
+import static org.junit.Assert.assertEquals;
+
+public class FibonacciTest {
+
+    Fibonacci target = new Fibonacci();
+
+    @Test
+    public void shouldBe1() {
+        shouldBe(1, 1);
```

Comment:
I would argue that one test per test case here is over-engineering. I would write one test method and do all my checks inside it, personally. There's purity, and then there's pragmatism. In this case I'd say being pragmatic (one test method) improves the maintainability and readability of the test.

Keep using the extracted `shouldBe` method though.